src/components/Pinpad/Pinpad.tsx

```diff
@@ -0,0 +1,100 @@
+import { useState, useEffect } from "react";
+import { ErrorDiv } from "../ErrorDiv/ErrorDiv";
+
+import styles from "./Pinpad.module.css";
+import {
+	PINPAD_MUI_KEYS,
+	LINK_ID_LENGTH,
+	VITESSCE_LINK_SITE,
+} from "../../utils/constants";
+import { exampleConfigHeadset } from "../../utils/config-examples";
+import { PinpadKey } from "../Pinpadkey";
+
+type ErrorType = string | null;
+
+export const Pinpad = () => {
+	const [linkId, setLinkId] = useState<string>("");
+	const [error, setError] = useState<ErrorType>(null);
+
+	const handleKeyPress = (key: string) => {
+		if (key === PINPAD_MUI_KEYS.BACKSPACE) {
+			setLinkId(() => linkId.slice(0, -1));
+		} else if (key === PINPAD_MUI_KEYS.DONE) {
+			attemptLogin();
+		} else if (linkId.length < LINK_ID_LENGTH && !isNaN(Number(key))) {
+			setLinkId(() => linkId + key);
+		}
+	};
+
+	// To remove the error as soon as we hit the digit limit
+	useEffect(() => {
+		if (linkId.length === LINK_ID_LENGTH) {
+			setError(null);
+		}
+	}, [linkId]);
+
+	const attemptLogin = () => {
+		// TODO: Do we want to put other checks in place, such as matching linkId?
+		if (linkId.length === LINK_ID_LENGTH) {
+			setError(() => null);
+			const layoutItem = exampleConfigHeadset.layout[0];
+			if (layoutItem && layoutItem.props) {
+				layoutItem.props.linkId = linkId;
+			}
+			const conf = JSON.stringify(exampleConfigHeadset, null, 2);
+			const nextUrl = `data:,${encodeURIComponent(conf)}`;
+			window.location.href = `${VITESSCE_LINK_SITE}${nextUrl}`;
+		} else {
+			setError(() => "Enter a valid Link ID");
```

Comment:
If the previous state is not used, we can omit the `() =>` part of the setter:

```suggestion
			setError("Enter a valid Link ID");
```